### ISSUE TITLE
[Question] Catching when the dropdown arrow is clicked

### ISSUE THREAD

**Issue by Flakmyr** [Blazored/Typeahead#240]
I'm using "@onfocus" to detect when the BlazoredTypeahead component gets focus, but that event does not get fired when the dropdown arrow (EnableDropDown="true") is clicked. How can I detect when the dropdown arrow is clicked?

**Comment by chrissainty** [comment]
I don't think this event works on a component. I think it will only work on HTML elements. What are you looking to do when the Typeahead gets focus?

**Comment by Flakmyr** [comment]
I'm using it to set a class to change the border color of the area. It works, except for when the dropdown arrow is clicked. But perhaps there is some other way to achieve this?

`<div class="component-@(_isSearch1?"active":"inactive")">`
`<BlazoredTypeahead @onfocus="OnFocus" EnableDropDown="true" ... </BlazoredTypeahead>`
`void OnFocus(FocusEventArgs args) { _isSearch1 = true; }`

**Comment by chrissainty** [comment]
Have you tried using the `:focus-within` pseudo selector? This is designed for this exact purpose and you wouldn't need to manually apply and remove a CSS class. There is a good overview of the selector [here](https://css-tricks.com/almanac/selectors/f/focus-within/)

**Comment by Flakmyr** [comment]
That's really nice, but I actually also need to toggle a radio button, so I don't think I can do without an actual event?

**Comment by chrissainty** [comment]
Ah, fair point. In that case we'd need to add a new event that could be subscribed to for this type of situation.

Just to set expectations, this probably won't be very quick. I'm finishing off my Blazor book right now and everything is taking a back seat to that. Once that is done I'll be overhauling a lot of these Blazored libraries as they've gotten a little neglected over the past year and a half and that is when I'd look to add this.
